plugins/resource_monitor_plugin/test/test_monitor_loop.cpp

```diff
@@ -0,0 +1,163 @@
+         return fixture.mock_get_space(p, ec);
+      }
+
+      space_handler_fixture& fixture;
+   };
+
+   boost::asio::io_context ctx;
+
+   using file_space_handler_t = file_space_handler<mock_space_provider>;
+   space_handler_fixture()
+   : space_handler(mock_space_provider( *this ), ctx)
+   {
+   }
+
+   void add_file_system(const bfs::path& path_name) {
+      space_handler.add_file_system( path_name );
+   }
+
+   void set_threshold(uint32_t threshold, uint32_t warning_threshold) {
+      space_handler.set_threshold( threshold, warning_threshold );
+   }
+
+   void set_sleep_time(uint32_t sleep_time) {
+      space_handler.set_sleep_time( sleep_time );
+   }
+
+   void set_shutdown_on_exceeded(bool shutdown_on_exceeded) {
+      space_handler.set_shutdown_on_exceeded(shutdown_on_exceeded);
+   }
+
+   bool is_threshold_exceeded() const {
+      return space_handler.is_threshold_exceeded();
+   }
+
+   int space_monitor_loop() {
+      return space_handler.space_monitor_loop();
+   }
+
+   bool test_loop_common(int num_loops, int interval)
+   {
+      mock_get_space = [ i = 0, num_loops ]( const bfs::path& p, boost::system::error_code& ec) mutable -> bfs::space_info {
+         ec = boost::system::errc::make_error_code(errc::success);
+
+         bfs::space_info rc;
+         rc.capacity  = 1000000;
+
+         if ( i < num_loops + 1 ) {  // "+ 1" for the get_space in add_file_system
+            rc.available = 300000;
+         } else {
+            rc.available = 100000;
+         }
+
+         i++;
+
+         return rc;
+      };
+
+      mock_get_stat = []( const char *path, struct stat *buf ) -> int {
+         buf->st_dev = 0;
+         return 0;
+      };
+
+      set_threshold(80, 75);
+      set_shutdown_on_exceeded(true);
+      set_sleep_time(interval);
+      add_file_system("/test");
+
+      auto start = std::chrono::system_clock::now();
+
+      auto monitor_thread = std::thread( [this] {
+         space_monitor_loop();
+         ctx.run();
+      });
+
+      monitor_thread.join();
+
+      auto end = std::chrono::system_clock::now();
+      std::chrono::duration<double> diff = end - start;
+
+      // As long as the thread ends within a margin (2 seconds for now)
+      // of the product of the number of loops by interval per loop,
+      // test is considered successful.
+      bool finished_in_time = (diff >= std::chrono::duration<double>(num_loops * interval) && diff <= std::chrono::duration<double>(num_loops * interval + 2)) ? true : false;
+
+      return finished_in_time;
+   }
+
+   // fixture data and methods
+   std::function<bfs::space_info(const bfs::path& p, boost::system::error_code& ec)> mock_get_space;
+   std::function<int(const char *path, struct stat *buf)> mock_get_stat;
+
+   file_space_handler_t space_handler;
+};
+
+BOOST_AUTO_TEST_SUITE(monitor_loop_tests)
+   BOOST_FIXTURE_TEST_CASE(zero_loop, space_handler_fixture)
+   {
+      BOOST_TEST(test_loop_common(0, 1)  == true);
```

Comment:
No need for the ```== true```.  BOOST_TEST is assuming a boolean value.

Reply:
Wanted to signify actual value and expected value. Will change.